src/components/filters/DateFilter/DateRangePicker/DateRangePicker.tsx

```diff
@@ -0,0 +1,194 @@
+                )}
+                placeholder={intl.formatMessage({ id: "filters.to" })}
+                // tslint:disable-next-line:jsx-no-lambda
+                onChange={event => this.handleToChange(toDateFromString(event.target.value))}
+                value={toInputDateString(to)}
+            />
+        ) : (
+            <DateRangePickerInputField
+                className={cx("s-date-range-picker-to", errorTo && "gd-date-range-picker-input-error")}
+                classNameCalendar="s-date-range-calendar-to"
+                ref={this.toInputRef}
+                onDayChange={this.handleToChange}
+                value={to || ""}
+                format={dateFormat}
+                placeholder={intl.formatMessage({ id: "filters.to" })}
+                dayPickerProps={{
+                    ...dayPickerPropsWithDefaults,
+                    onDayClick: this.handleToDayClick,
+                }}
+            />
+        );
+
+        return (
+            <>
+                <div className="gd-date-range-picker s-date-range-picker">
+                    {FromField}
+                    <span className="gd-date-range-picker-dash">&mdash;</span>
+                    {ToField}
+                </div>
+                {(errorFrom || errorTo) && (
+                    <DateRangePickerError
+                        errorId={
+                            // This means that when both inputs are invalid, error is shown only for "from"
+                            errorFrom || errorTo
+                        }
+                    />
+                )}
+            </>
+        );
+    }
+
+    private focusField = (inputRef: any) => {
+        if (inputRef.current) {
+            // Focus needs to happen on the next tick otherwise the day picker is not updated
+            setTimeout(() => {
+                inputRef.current.getInput().focus();
+            }, 0);
+        }
+    };
+
+    private blurField = (inputRef: any) => {
+        if (inputRef.current) {
+            inputRef.current.getInput().blur();
+        }
+    };
+
+    private updateRange = (from: Date, to: Date) => {
+        this.props.onRangeChange({ from, to });
+    };
+
+    private handleFromDayClick = () => {
+        this.focusField(this.toInputRef);
+    };
+
+    private handleToDayClick = (to: Date) => {
+        const rangeBoundsCrossed = areRangeBoundsCrossed(this.props.range.from, to);
+        if (to && !rangeBoundsCrossed) {
+            this.blurField(this.toInputRef);
+        } else if (rangeBoundsCrossed) {
+            this.focusField(this.fromInputRef);
+        }
+    };
+
+    private handleFromChange = (from: Date) => {
+        const to =
+            from && this.props.range.to && areRangeBoundsCrossed(from, this.props.range.to)
+                ? from
+                : this.props.range.to;
+
+        const inputValue = this.fromInputRef.current && this.fromInputRef.current.getInput().value;
```

Comment:
I did small refactor, but I don't want to touch the code more since it is not clear to me how this works and it is not tested... Don't want to break the thin ice ;)